spec/support/saml_auth_helper.rb

```diff
@@ -49,280 +55,88 @@ def idp_fingerprint
-    settings
-  end
-
-  def sp_not_requesting_signed_saml_response_settings
-    settings = saml_settings.dup
-    settings.issuer = 'test_saml_sp_not_requesting_signed_response_message'
-    settings
-  end
-
-  def sp_requesting_signed_saml_response_settings
-    settings = saml_settings.dup
-    settings.issuer = 'test_saml_sp_requesting_signed_response_message'
-    settings
-  end
-
-  def email_nameid_saml_settings
-    settings = saml_settings.dup
-    settings.name_identifier_format = Saml::Idp::Constants::NAME_ID_FORMAT_EMAIL
-    settings
-  end
-
-  def missing_nameid_format_saml_settings
-    settings = saml_settings.dup
-    settings.name_identifier_format = nil
-    settings
-  end
-
-  def email_nameid_saml_settings_for_disallowed_issuer
-    settings = saml_settings.dup
-    settings.name_identifier_format = Saml::Idp::Constants::NAME_ID_FORMAT_EMAIL
-    settings
-  end
-
-  def ial1_saml_settings
-    settings = saml_settings.dup
-    settings.authn_context = Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF
-    settings
-  end
-
-  def sp1_ial1_saml_settings
-    settings = sp1_saml_settings.dup
-    settings.authn_context = Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF
-    settings
-  end
-
-  def sp1_ial2_saml_settings
-    settings = sp1_saml_settings.dup
-    settings.authn_context = Saml::Idp::Constants::IAL2_AUTHN_CONTEXT_CLASSREF
-    settings
-  end
-
-  def sp1_ial_max_saml_settings
-    settings = sp1_saml_settings.dup
-    settings.authn_context = Saml::Idp::Constants::IALMAX_AUTHN_CONTEXT_CLASSREF
-    settings
-  end
-
-  def sp1_ial2_strict_saml_settings
-    settings = sp1_saml_settings.dup
-    settings.authn_context = Saml::Idp::Constants::IAL2_STRICT_AUTHN_CONTEXT_CLASSREF
-    settings
-  end
-
-  def loa3_saml_settings
-    settings = sp1_saml_settings.dup
-    settings.authn_context = Saml::Idp::Constants::LOA3_AUTHN_CONTEXT_CLASSREF
-    settings
-  end
-
-  def ialmax_saml_settings
-    settings = sp1_saml_settings.dup
-    settings.authn_context = Saml::Idp::Constants::IALMAX_AUTHN_CONTEXT_CLASSREF
-    settings
-  end
-
-  def ialmax_with_bundle_saml_settings
-    settings = ialmax_saml_settings
-    settings.authn_context = [
-      settings.authn_context,
-      "#{Saml::Idp::Constants::REQUESTED_ATTRIBUTES_CLASSREF}first_name:last_name email, ssn",
-      "#{Saml::Idp::Constants::REQUESTED_ATTRIBUTES_CLASSREF}phone",
-    ]
-    settings
+  def logout_request
+    @logout_request ||= OneLogin::RubySaml::Logoutrequest.new
   end
 
-  def ial2_with_bundle_saml_settings
-    settings = sp1_ial2_saml_settings
-    settings.authn_context = [
-      settings.authn_context,
-      "#{Saml::Idp::Constants::REQUESTED_ATTRIBUTES_CLASSREF}first_name:last_name email, ssn",
-      "#{Saml::Idp::Constants::REQUESTED_ATTRIBUTES_CLASSREF}phone",
-    ]
-    settings
+  def saml_authn_request_url(saml_overrides: {}, saml_security_overrides: {}, params: {})
+    @saml_authn_request = auth_request.create(
```

Comment:
Do we ever reference this instance variable?

```suggestion
    auth_request.create(
```

Reply:
It is saved off in several instances, e.g., [this example saves the request, then compares against it near the end](https://github.com/18F/identity-idp/blob/main/spec/features/saml/ial1_sso_spec.rb#L7-L37)